preprocess_utils.py

```diff
@@ -0,0 +1,271 @@
+          "words in each shortened sentence that are not in the original sentence.",
+          "( std:", "{:.2f}".format(np.std(count_diff)), ")")
+
+    # average compression ratio
+    compression_ratio = []
+    for i in range(len(sentences)):
+        compression_ratio.append(len(summaries[i].split()) / len(sentences[i].split()))
+    compression_ratio = np.array(compression_ratio)
+    print("The average compression ratio is", "{:.2f}".format(np.mean(compression_ratio)), "( std:",
+          "{:.2f}".format(np.std(compression_ratio)), ")")
+
+
+def tokenize_with_space(sentences):
+    """Tokenize the input text with spaces separating the tokens.
+    Args:
+      sentences: a column of input sentences.
+    
+    Returns:
+      a column of tokenized texts with spaces separating the tokens.
+    """
+    spaced_sentences = []
+    for sentence in sentences:
+        tokens = nltk.word_tokenize(sentence)
+        spaced_sentence = ""
+        for j in range(len(tokens) - 1):
+            spaced_sentence += tokens[j] + " "
+        spaced_sentence += tokens[-1]
+        spaced_sentences.append(spaced_sentence)
+
+    return spaced_sentences
+
+
+def split_dataset(train_path, tune_path, valid_path, preprocessed_path, num_of_tuning_sam, num_of_valid_sam,
+                  whether_shuffle):
+    """Split the dataset into training, tuning, and validation sets, and store each in a file.
+        Args:
+          train_path: path to store the training set.
+          tune_path: path to store the tuning set.
+          valid_path: path to store the validation set.
+          preprocessed_path: path where the preprocessed dataset is store.
+          num_of_tuning_sam: number of tuning samples.
+          num_of_valid_sam: number of validation samples.
+          whether_shuffle: whether the dataset needs to be shuffled before splitting.
+
+        Side effects:
+          Training set is stored in the path specified by train_path.
+          Tuning set is stored in the path specified by tune_path.
+          Validation set is stored in the path specified by valid_path.
+    """
+    sentences = []
+    summaries = []
+    with open(os.path.expanduser(preprocessed_path), 'r') as f:
+        tsv_reader = csv.reader(f, delimiter='\t')
+        for line in tsv_reader:
+            sentences.append(line[0])
+            summaries.append(line[1])
+
+    validate_dataset(sentences, summaries)
+
+    sentence_shuffled = []
+    summary_shuffled = []
+
+    if num_of_tuning_sam + num_of_valid_sam > len(sentences):
+        raise Exception("The number of tuning and validation samples together exceeds the total sample size of " + str(
+            len(sentences)))
+
+    if whether_shuffle:
+        print("-------Shuffling the dataset-------")
+        sentence_shuffled = []
+        summary_shuffled = []
+        index_shuffled = list(range(len(sentences)))
+        random.shuffle(index_shuffled)
+    else:
```

Comment:
If whether_shuffle is false, maybe you can just leave whole data not suffle

Reply:
Done. I added another flag to indicate whether each file needs to be shuffled.